spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -131,6 +131,12 @@ public int orcBatchSize() {
         .parse();
   }
 
+  public Long splitSizeOption() {
```

Comment:
Should this replace `splitSize` instead of adding a parallel call? The `SparkReadConf` is not yet released, so we can change it still.

Reply:
Ah, nevermind. I see that the other `splitSize` method merges in the table settings. I think that's fine for now, but in a follow-up we should start building the Iceberg `TableScan` in the scan builder and passing it in. Then we could use the new `TableScan` methods everywhere instead of relying on `SparkReadConf` to duplicate logic to produce those settings.

Reply:
I am not a fan of this either. I tried that but the main problem is `SparkFilesScan`. It consumes tasks from a cache instead of planning. Any ideas how to handle it, @rdblue?

Reply:
I think we'll probably move `SparkFilesScan` over to getting the config setting for a scan as well, so let's not worry about it for now.